### ISSUE TITLE
iohook\builds\electron-v64-win32-ia32\build\Release\iohook.node is not a valid Win32 application

### ISSUE THREAD

**Issue by gvahuja** [wilix-team/iohook#140]
## Expected Behavior
The electron app should work properly on both 64 bit and 32 bit machines

## Current Behavior
The app is working properly on 64 bit machine
Not working on 32 bit machine getting following error

[Screenshot] (https://github.com/gvahuja/electron-issue/blob/master/Screenshot%20(3).png)

## Possible Solution
I thought the possible solutions would be:
1. Compile electron-app and build executable for 32 bit in 32 bit machine (didn't work)
2. Copy the 32 bit binaries of iohook from [here](https://github.com/wilix-team/iohook/releases) and paste them in the respective build folder to work. (not working)

## Context
I want to run my electron app in 32 bit windows machine
My package.json config for iohook

`
"iohook": {
    "targets": [
      "node-64",
      "electron-64"
    ],
    "platforms": [
      "win32",
      "darwin",
      "linux"
    ],
    "arches": [
      "x64",
      "ia32"
    ]
  }`

## Your Environment
* Version used: iohook 0.3.1
* Environment name and version: node-v64 electron-v64
* Operating System and version (desktop or mobile): Windows 10 x64 and Windows 7 x86

**Comment by gvahuja** [comment]
@Djiit Can you help?

**Comment by tbs-arpit** [comment]
Same error here. 

It works in debug mode (like when I do `electron main.js`) but when I package it using electron-packager into an exe then it is giving me this error 

![image](https://user-images.githubusercontent.com/41741151/51990304-c5afb380-24ce-11e9-9cb1-db9fee958380.png)

### Here is my package.json 

<details><summary>Package.json</summary>
<p>

```json
{
  "name": "<Project name :P>",
  "version": "1.19.0",
  "description": "A simple electron app for <project name :P>",
  "main": "main.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "electron main.js",
    "setup": "node build.js",
    "icon": "node icon.js"
  },
  "repository": {
    "type": "git",
    "url": "<Project URL>"
  },
  "iohook": {
    "targets": [
      "node-59",
      "electron-54"
    ],
    "platforms": [
      "win32",
      "darwin"
    ],
    "arches": [
      "x64",
      "ia32"
    ]
  },
  "cmake-js": {
    "runtime": "electron",
    "runtimeVersion": "1.4.13"
  },
  "keywords": [
    "electron",
    "windows",
    "mac"
  ],
  "author": "Arpit Vasani",
  "license": "ISC",
  "devDependencies": {
    "asar": "^0.14.0",
    "electron": "^1.7.10",
    "electron-installer-dmg": "^0.2.1",
    "electron-installer-windows": "*",
    "electron-packager": "^10.1.1",
    "electron-winstaller": "^2.6.3"
  },
  "dependencies": {
    "aws-sdk": "^2.340.0",
    "electron-rebuild": "^1.8.2",
    "electron-squirrel-startup": "^1.0.0",
    "iohook": "^0.3.1",
    "network": "^0.4.0",
    "util": "^0.11.1"
  }
}
```

</p>
</details>

**Comment by tbs-arpit** [comment]
@Djiit 

Found some pointers which should help reproducing this issue

I took electron example and installed electron-packager and here is what I did 

✔️ set iohook arches to x64
✔️ set electron-packager arch to x64
✔️ OS is x64

Made an exe works like a charm **Boom!** 💥 💥 

then I set the following things 

✔️ set iohook arches to ia32
✔️ set eletron-packager arch to ia32
✔️  OS is x64

and the's where I got his beauty in windows dialog

```
[Window Title]
Error

[Main Instruction]
A JavaScript error occurred in the main process

[Content]
Uncaught Exception:
Error: %1 is not a valid Win32 application.
<project path>\electron-example\electron-example-win32-ia32\resources\app\node_modules\iohook\builds\electron-v57-win32-ia32\build\Release\iohook.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (E:\Arpit\electron-example\electron-example-win32-ia32\resources\app\node_modules\iohook\index.js:10:21)
    at Object.<anonymous> (E:\Arpit\electron-example\electron-example-win32-ia32\resources\app\node_modules\iohook\index.js:397:3)

[OK]
```


**Now as per [this link](https://github.com/electron/electron/issues/1075#issuecomment-72330772) it could be 2 reasons why this is happening** 

Please check if that is the case.

**Comment by Djiit** [comment]
As stated in another issue (can't find it sadly), I'm not sure we have a correct build for 32bit version. Is there any travis expert that can check our build job file ?

**Comment by gvahuja** [comment]
I referred [this comment](https://github.com/wilix-team/iohook/issues/70#issuecomment-412623935) and build executable with electron-v57 it worked for windows 32-bit machine also but I want to use electron-v64. I'll also try the same steps for v64 as the author of [this comment](https://github.com/wilix-team/iohook/issues/70#issuecomment-412623935) has mentioned for v-57 and will comment if it works.

**Comment by gvahuja** [comment]
I was able to build win32 ia32 binaries for electron v3.0.0 (ABI v64) and node v10.14.1 (ABI v64). I just followed the steps of [this comment](https://github.com/wilix-team/iohook/issues/70#issuecomment-412623935).
[Here](https://github.com/gvahuja/iohook-prebuild-v64-win32-ia32) are the builds, follow the steps in Readme to resolve the issue. I am using the builds and it's working properly on Windows 32bit machine.
